enclave_manager/avalon_enclave_manager/graphene/graphene_enclave_manager.py

```diff
@@ -0,0 +1,199 @@
+
+    def _manager_on_boot(self):
+        """
+        Executes Boot flow of enclave manager
+        """
+        logger.info("Executing boot time procedure")
+
+        # Add a new worker
+        worker_info = EnclaveManager.create_json_worker(self, self._config)
+        # Hex string read from config which is 64 characters long
+        worker_id = self._worker_id
+        self._worker_kv_delegate.add_new_worker(worker_id, worker_info)
+        # Update mapping of worker_id to workers in a pool
+        self._worker_kv_delegate.update_worker_map(
+            worker_id, self._identity)
+
+        # Cleanup all stale work orders for this worker which
+        # used old worker keys
+        self._wo_kv_delegate.cleanup_work_orders()
+
+# -------------------------------------------------------------------------
+
+    def _create_signup_data(self):
+        """
+        Creates Graphene worker signup data.
+
+        Returns :
+            signup_data: Signup data containining information of worker
+                         like public encryption and verification key,
+                         worker quote, mrencalve
+        """
+        json_request = self._create_json_request("ProcessWorkerSignup", None)
+        # Send signup request to Graphene worker
+        worker_signup = self.zmq_socket.send_request_zmq(
+                                        json.dumps(json_request))
+        if worker_signup is None:
+            logger.error("Unable to get Graphene worker signup data")
+            return None
+        logger.debug("Graphene signup result {}".format(worker_signup))
+        try:
+            worker_signup_json = json.loads(worker_signup)
+        except json.decoder.JSONDecodeError:
+            logger.error("Invalid Json: Graphene worker signup data")
+            return None
+        # Create Signup Graphene object
+        signup_data = SignupGraphene(worker_signup_json)
+        return signup_data
+
+# -------------------------------------------------------------------------
+
+    def _execute_wo_in_trusted_enclave(self, input_json_str):
+        """
+        Submits workorder request to Graphene Worker and retrieves
+        the response
+
+        Parameters :
+            input_json_str: JSON formatted str of the request to execute
+        Returns :
+            JSON response received from Graphene worker.
+        """
+        json_request = self._create_json_request("ProcessWorkOrder",
+                                                 input_json_str)
+        result = self.zmq_socket.send_request_zmq(json.dumps(json_request))
+        try:
+            json_response = json.loads(result)
+        except json.decoder.JSONDecodeError:
+            logger.error("Invalid Json: Graphene execution result")
+            return None
+        return json_response
+
+# -------------------------------------------------------------------------
+
+    def _create_json_request(self, method_name, params=None):
```

Comment:
Can we move this utility function to common python module?

Reply:
We ideally should. May be a good place will be to keep this function is in https://github.com/hyperledger/avalon/blob/master/common/python/utility/jrpc_utility.py
But it would require modify existing file which wanted to avoid at this stage of release. I can add a TODO